modules/externalipam/ipam_api.rb

```diff
@@ -0,0 +1,389 @@
+      end
+    end
+
+    # Get a list of subnets for a given External IPAM group
+    #
+    # Input:   1. group:         The name of the External IPAM group
+    #
+    # Returns:
+    #   Response if success:
+    #   ===========================
+    #     Http Code:     200
+    #     JSON Response:
+    #       {"data":[
+    #         {"subnet":"10.20.30.0","mask":"29","description":"This is a subnet"},
+    #         {"subnet":"40.50.60.0","mask":"29","description":"This is another subnet"}
+    #       ]}
+    #
+    #   Response if no subnets exist in group:
+    #   ===========================
+    #     Http Code:     404
+    #     JSON Response:
+    #       {"error": "No subnets found in External IPAM group"}
+    #
+    #   Response if groups are not supported:
+    #   ===========================
+    #     Http Code:     500
+    #     JSON Response:
+    #       {"error": "Groups are not supported"}
+    get '/groups/:group/subnets' do
+      content_type :json
+
+      begin
+        validate_required_params!([:group], params)
+
+        group_name = get_request_group(params)
+        subnets = provider.get_ipam_subnets(group_name)
+
+        halt 404, { error: ERRORS[:no_subnets_in_group] }.to_json if subnets.nil?
+        subnets.to_json
+      rescue Proxy::Validations::Error => e
+        logger.warn(e.message)
+        halt 400, { error: e.to_s }.to_json
+      rescue RuntimeError => e
+        logger.warn(e.message)
+        halt 500, { error: e.message }.to_json
+      rescue Errno::ECONNREFUSED, Errno::ECONNRESET
+        logger.warn(ERRORS[:no_connection])
+        halt 500, { error: ERRORS[:no_connection] }.to_json
+      end
+    end
+
+    # Checks whether an IP address has already been taken in External IPAM
+    #
+    # Inputs: 1. address:         The network address of the IPv4 or IPv6 subnet.
+    #         2. prefix:          The subnet prefix(e.g. 24)
+    #         3. ip:              IP address to be queried
+    #         4. group(optional): The name of the External IPAM Group, containing the subnet to check
+    #
+    # Returns:
+    #   Response if exists:
+    #   ===========================
+    #     Http Code:  200
+    #     Response:   true
+    #
+    #   Response if not exists:
+    #   ===========================
+    #     Http Code:      200
+    #     JSON Response:  false
+    get '/subnet/:address/:prefix/:ip' do
+      content_type :json
+
+      begin
+        validate_required_params!([:address, :prefix, :ip], params)
+
+        ip = validate_ip!(params[:ip])
+        cidr = validate_cidr!(params[:address], params[:prefix])
+        group_name = get_request_group(params)
+        subnet = provider.get_ipam_subnet(cidr, group_name)
+
+        halt 404, { error: ERRORS[:no_subnet] }.to_json if subnet.nil?
+        validate_ip_in_cidr!(ip, cidr)
+        ip_exists = provider.ip_exists?(ip, subnet[:id], group_name)
+        halt 200, ip_exists.to_json
```

Comment:
Ditto:

```suggestion
        halt 200, {result: ip_exists}.to_json
```

Reply:
This is a typical problem - later on we would like to extend this with an error message explaining maybe a reason why the IP was taken (where it belongs to etc).